frontend/src/types/constants.ts

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+// with the License. A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0/
+//
+// or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+export enum NodeType {
+  HeadNode = "HeadNode",
+  ComputeNode = "ComputeNode",
+}
+
+export enum InstanceState {
+  Pending = "pending",
+  Running = "running",
+  ShuttingDown = "shutting-down",
+  Terminated = "terminated",
+  Stopping = "stopping",
+  Stopped = "stopped",
+  }
+
+export enum ComputeFleetStatus {
+  StartRequested = "START_REQUESTED",  // works only with Slurm
+  Starting = "STARTING",  // works only with Slurm
+  Running = "RUNNING",  // works only with Slurm
+  Protected = "PROTECTED",  // works only with Slurm
+  StopRequested = "STOP_REQUESTED",  // works only with Slurm
+  Stopping = "STOPPING",  // works only with Slurm
+  Stopped = "STOPPED",  // works only with Slurm
+  Unknown = "UNKNOWN",  // works only with Slurm
+  Enabled = "ENABLED",  // works only with AWS Batch
+  Disabled = "DISABLED",  // works only with AWS Batch
+}
```

Comment:
suggestion: I'm not sure I'm getting what's happening here. Are you merging two different types? If so, I'd find a union type (or other type merging strategy) a bit more robut.

I'm thinking something along these lines:

```ts
export type ComputeFleetStatus = SlurmFleetStatus | AWSBatchFleetStatus
```

This is [not easily done](https://stackoverflow.com/a/55827534) with `enum`s, but there could be a solution. 

Probably union types are the better choice here, since we're are basically defining types, but I don't have strong opinions against enums (however here's a [fairly short read against enums](https://fettblog.eu/tidy-typescript-avoid-enums/) 😁 )

Reply:
This is actually how it's defined directly from the types in ParallelCluster. So I might prefer to keep things consistent with that -- I'm happy to break it out which might make our typing easier in certain cases (?), could also be broken out when needed.